.github/workflows/codeql.yml

```diff
@@ -0,0 +1,33 @@
+name: CodeQL
+on:
+  pull_request:
+    paths:
+      - .github/workflows/*
+      - .github/codeql.yml
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: 0 2 * * *
+  workflow_dispatch: ~
```

Comment:
Since the CodeQL version and config is(/should be) fixed for a given commit, it seems unnecessary to run this on a recurring schedule. 

```suggestion
```